contrib/babelfishpg_tsql/src/catalog.c

```diff
@@ -2265,3 +2266,42 @@ create_guest_role_for_db(char *dbname)
 	/* Set current user back to previous user */
 	bbf_set_current_user(prev_current_user);
 }
+
+/*
+ * Retrieve the db_owner role name of a specific
+ * database from the catalog, it doesn't rely on the
+ * migration mode GUC.
+ */
+static char *
+get_db_owner_role_name(char *dbname)
+{
+	Relation	bbf_authid_user_ext_rel;
+	HeapTuple	tuple_user_ext;
+	ScanKeyData		key[2];
+	TableScanDesc		scan;
+	char		*db_owner_role;
```

Comment:
Let's initialize `db_owner_role` with NULL so that we don't return an uninitialized pointer if tuple_user_ext is somehow invalid.